GeometryCore/src/Polygon.cpp

```diff
@@ -177,9 +177,27 @@ namespace geom_utils
     // For non-convex polygons it should return just empty vector.
     std::vector<Triangle2D> Polygon::triangulate() const
     {
-        return {}; 
+        std::vector <Triangle2D> vectorForTriangulate;
+
+     
+        if (points.size() >= 4) {
+            if (isConvexHull())
+            {
+           
+                FPoint2D  firstVertex =  points[0];
+                for (int i = 0; i < points.size() - 2 ; i++)
+                {
+                   
+                    Triangle2D tri = {points[(i + 2) % points.size()] ,firstVertex, points[(i+1) % points.size()] };
+                    vectorForTriangulate.push_back(tri);
+                }
+            }
+            return vectorForTriangulate;
```

Comment:
What method should return if points.size() < 4?

Reply:
Does he also have to return an empty vector?

Reply:
Lets think. In general, we can have the following input:
- empty polygon or 1- or 2- points polygon
- 3 or more points polygon

Please think what output should be in those cases.

Please analyse carefully **input** and **output** each time while developing some method/function etc.